hudi-common/src/main/java/org/apache/hudi/common/table/log/HoodieMergedLogRecordScanner.java

```diff
@@ -445,7 +444,7 @@ public Builder withOptimizedLogBlocksScan(boolean enableOptimizedLogBlocksScan)
 
     @Override
     public Builder withRecordMerger(HoodieRecordMerger recordMerger) {
-      this.recordMerger = HoodieRecordUtils.mergerToPreCombineMode(recordMerger);
+      this.recordMerger = recordMerger;
```

Comment:
Does the merger implements `OperationModeAwareness` then?

Reply:
I think it does, the HoodieAvroRecordMerger.

Reply:
The `#preCombine` is invoked for log records merging in the very first place for early releases, let's figure out why the compaction got issues, at least the `#preCombine` also choose records based on `orderingVal`.

Reply:
I think it becomes an issue since we are trying to enforce commit time merging. Before this, we did not enforce consistent behavior between ingestion and compaction/clustering.